### ISSUE TITLE
Think about packaging

### ISSUE THREAD

**Issue by andrewpatto** [umccr/cloud-checksum#11]
_Were_ this to become a popular/useful utility - how does it get published/available in distros? Brew etc?

I presume we can start with a brew tap immediately and host it via the repo?

**Comment by alexiswl** [comment]
I also wouldn't be surprised if this doesn't work particularly well with rust. Any sort of compiling I've found can take a very long time with QEMU / cross-compilation methods.

**Comment by brainstorm** [comment]
> I also wouldn't be surprised if this doesn't work particularly well with rust. Any sort of compiling I've found can take a very long time with QEMU / cross-compilation methods.

Yeah, I also share that experience... perhaps I was trying to go with matrix (and accidentally left it halfway configured in GHA) because I was trying to compile it with separate runners, but then it becomes a problem because one needs a single image containing several architectures (like OSX "fat" binaries, as an analogy).

**Comment by alexiswl** [comment]
Looking through https://github.com/umccr/copyrite/blob/package/pkg/docker/Dockerfile, and the time spent doing the rust compiling 1hr+ (https://github.com/umccr/copyrite/actions/runs/19024092629) vs making the binary assets 13 mins (https://github.com/umccr/copyrite/actions/runs/19024092637). 

Why not just do the docker build AFTER the binary releases GH action, using the outputs of the binary releases? Saves having to do a second cargo-build per architecture. Becomes merely a 

```
FROM debian

ARG TARGETPLATFORM
ARG GITHUB_REPO_RELEASES_URL
ARG PACKAGE_VERSION

# Install debian basics
RUN \
    # Set boolean \
    if [ "${TARGETPLATFORM#linux/}" = "arm64" ]; then \
      platform_url="aarch64";  \
    else \
      platform_url="x86_64"; \
    fi && \
    # Install
    echo "Standard APT" 1>&2 && \
    apt update -yq && \
    apt upgrade -yq && \
    apt install -yq \
      wget \
    wget \
      --quiet \
      --output-document - \
    "${GITHUB_REPO_RELEASES_URL}/${PACKAGE_VERSION}/cloud-checksum-${platform_url}-linux-gnu.tar.gz" | \
    tar -C /usr/bin -xf - \
    chmod +x /usr/bin/cloud-checksum

```
